modules/shared/src/main/scala/org/tessellation/schema/GlobalSnapshot.scala

```diff
@@ -24,6 +30,40 @@ import derevo.derive
 import eu.timepit.refined.auto._
 import eu.timepit.refined.types.numeric.PosInt
 
+@derive(eqv, show, encoder, decoder)
+case class GlobalIncrementalSnapshot(
+  ordinal: SnapshotOrdinal,
+  height: Height,
+  subHeight: SubHeight,
+  lastSnapshotHash: Hash,
+  blocks: SortedSet[BlockAsActiveTip[DAGBlock]],
+  stateChannelSnapshots: SortedMap[Address, NonEmptyList[Signed[StateChannelSnapshotBinary]]],
+  rewards: SortedSet[RewardTransaction],
+  epochProgress: EpochProgress,
+  nextFacilitators: NonEmptyList[PeerId],
+  tips: SnapshotTips,
+  stateProof: MerkleTree
```

Comment:
Could you please explain in your own words the benefits of adding this field? 

Unfortunately, the paragraph about Merkle Tree in ADR #538 doesn't have any meaningful content.

Reply:
We split the state from the snapshot. State proof is a way to confirm that the state corresponds to the snapshot (or in other words, that the node properly calculated the state, when it's state proof matches the one from snapshot).
It's a merkle tree rather than a single hash (hash of the serialized state) because that way we can proof parts of the state separately (partial state). As an example: I can ask for just the balances and I can proof they are correct (with no need to fetch the whole state, so no need to download state channel contexts).

Reply:
Ok, but you're storing a whole tree here so why not just check if there's a leaf with a given hash?

Reply:
you need to store the tree somewhere to make proofs for partial checks.

Reply:
Ok, but why not just store the leaves? Why the whole tree?

If I want to check if my locally computed state is the same as the state computed on the nodes that created and signed the snapshot, then I can compare the hash of my state with the hash that's in the leaves. Am I missing something?

Reply:
Ok, then please either change it to the single hash or a list of hashes (leaves). This is important because changing it in the future will require a schema change.